src/nomad_material_processing/vapor_deposition/cvd/movpe.py

```diff
@@ -0,0 +1,215 @@
+    )
+
+
+class Movpe(VaporDeposition, EntryData):
+    """
+    Metal-organic Vapor Phase Epitaxy (MOVPE) is a chemical vapor deposition method
+    used to produce single- or multi-layered thin films.
+    """
+
+    m_def = Section(
+        a_eln=ELNAnnotation(
+            properties=SectionProperties(
+                order=[
+                    'name',
+                    'method',
+                    'datetime',
+                    'end_time',
+                    'duration',
+                ],
+            ),
+        ),
+        label_quantity='lab_id',
+    )
+    method = Quantity(
+        type=str,
+        default='MOVPE',
+    )
+    steps = SubSection(
+        section_def=StepMovpe,
+        repeats=True,
+    )
+
+    def normalize(self, archive, logger):
+        """
+        The workflow of the Movpe process archive is populated
+        with the input and output links.
+        """
+        archive.workflow2 = None
+        super().normalize(archive, logger)
+        if self.steps is not None:
+            inputs = []
+            outputs = []
+            for step in self.steps:
+                if step.sample_parameters is not None:
+                    for sample in step.sample_parameters:
+                        if sample.layer is not None:
+                            outputs.append(
+                                Link(
+                                    name=f'{sample.layer.name}',
+                                    section=sample.layer.reference,
+                                )
+                            )
+                        if sample.substrate is not None:
+                            outputs.append(
+                                Link(
+                                    name=f'{sample.substrate.name}',
+                                    section=sample.substrate.reference,
+                                )
+                            )
+                        if (
+                            sample.substrate is not None
+                            and sample.substrate.reference is not None
+                        ):
+                            if hasattr(
+                                getattr(sample.substrate.reference, 'substrate'),
+                                'name',
+                            ):
+                                inputs.append(
+                                    Link(
+                                        name=f'{sample.substrate.reference.substrate.name}',
+                                        section=getattr(
+                                            sample.substrate.reference.substrate,
+                                            'reference',
+                                            None,
+                                        ),
+                                    )
+                                )
+            archive.workflow2.outputs.extend(set(outputs))
+            archive.workflow2.inputs.extend(set(inputs))
```

Comment:
I guess this is something we should do in general for `VaporDeposition` but let's leave it here for now. I'm not sure about the need for the `hasattr` and `getattr` but for now I would just suggest to try and reduce the indentation level a bit (untested):
```suggestion
        if self.steps is None:
            return
        inputs = []
        outputs = []
        for step in self.steps:
            if step.sample_parameters is None:
                continue
            for sample in step.sample_parameters:
                if sample.layer is not None:
                    outputs.append(
                        Link(
                            name=f'{sample.layer.name}',
                            section=sample.layer.reference,
                        )
                    )
                if sample.substrate is not None:
                    outputs.append(
                        Link(
                            name=f'{sample.substrate.name}',
                            section=sample.substrate.reference,
                        )
                    )
                if (
                    sample.substrate is not None
                    and sample.substrate.reference is not None
                    and hasattr(
                        getattr(sample.substrate.reference, 'substrate'),
                        'name',
                    )
                ):
                    inputs.append(
                        Link(
                            name=f'{sample.substrate.reference.substrate.name}',
                            section=getattr(
                                sample.substrate.reference.substrate,
                                'reference',
                                None,
                            ),
                        )
                    )
        archive.workflow2.outputs.extend(set(outputs))
        archive.workflow2.inputs.extend(set(inputs))
```